src/abstract-interpretation/data-frame/mappers/function-mapper.ts

```diff
@@ -0,0 +1,1440 @@
+import { escapeRegExp, filterValidNames, getArgumentValue, getEffectiveArgs, getFunctionArgument, getFunctionArguments, getUnresolvedSymbolsInExpression, hasCriticalArgument, isDataFrameArgument, isNamedArgument, isRNull } from './arguments';
+
+/**
+ * Represents the different types of data frames in R
+ */
+enum DataFrameType {
+	DataFrame = 'data.frame',
+	Tibble = 'tibble',
+	DataTable = 'data.table'
+}
+
+/**
+ * Mapper for mapping the supported concrete data frame functions to mapper functions,
+ * including information about the origin library of the functions and the type of the returned data frame.
+ */
+const DataFrameFunctionMapper = {
+	'data.frame':    { mapper: mapDataFrameCreate },
+	'as.data.frame': { mapper: mapDataFrameConvert },
+	'read.table':    { mapper: mapDataFrameRead },
+	'read.csv':      { mapper: mapDataFrameRead },
+	'read.csv2':     { mapper: mapDataFrameRead },
+	'read.delim':    { mapper: mapDataFrameRead },
+	'read.delim2':   { mapper: mapDataFrameRead },
+	'read_table':    { mapper: mapDataFrameRead, library: 'readr', returnType: DataFrameType.Tibble },
+	'read_csv':      { mapper: mapDataFrameRead, library: 'readr', returnType: DataFrameType.Tibble },
+	'read_csv2':     { mapper: mapDataFrameRead, library: 'readr', returnType: DataFrameType.Tibble },
+	'read_tsv':      { mapper: mapDataFrameRead, library: 'readr', returnType: DataFrameType.Tibble },
+	'read_delim':    { mapper: mapDataFrameRead, library: 'readr', returnType: DataFrameType.Tibble },
+	'cbind':         { mapper: mapDataFrameColBind },
+	'rbind':         { mapper: mapDataFrameRowBind },
+	'head':          { mapper: mapDataFrameHeadTail },
+	'tail':          { mapper: mapDataFrameHeadTail },
+	'subset':        { mapper: mapDataFrameSubset },
+	'filter':        { mapper: mapDataFrameFilter, library: 'dplyr' },
+	'select':        { mapper: mapDataFrameSelect, library: 'dplyr' },
+	'mutate':        { mapper: mapDataFrameMutate, library: 'dplyr' },
+	'transform':     { mapper: mapDataFrameMutate },
+	'group_by':      { mapper: mapDataFrameGroupBy, library: 'dplyr', returnType: DataFrameType.Tibble },
+	'summarise':     { mapper: mapDataFrameSummarize, library: 'dplyr' },
+	'summarize':     { mapper: mapDataFrameSummarize, library: 'dplyr' },
+	'inner_join':    { mapper: mapDataFrameJoin, library: 'dplyr' },
+	'left_join':     { mapper: mapDataFrameJoin, library: 'dplyr' },
+	'right_join':    { mapper: mapDataFrameJoin, library: 'dplyr' },
+	'full_join':     { mapper: mapDataFrameJoin, library: 'dplyr' },
+	'merge':         { mapper: mapDataFrameJoin },
+	'relocate':      { mapper: mapDataFrameIdentity, library: 'dplyr' },
+	'arrange':       { mapper: mapDataFrameIdentity, library: 'dplyr' }
+} as const satisfies Record<string, DataFrameFunctionMapperInfo<never>>;
+
+/**
+ * List of data frame functions that are not explicitly supported but may return data frames.
+ */
+const UnsupportedDataFrameFunctions = [
```

Comment:
Unsupported sounds like something that any not explicitly stated function is. Maybe Just `Other`?